src/arch/src/aarch64/cache_info.rs

```diff
@@ -1,27 +1,43 @@
 // Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::fmt::{Display, Formatter};
-use std::path::{Path, PathBuf};
-use std::{fs, io, result};
+use std::{
+    any::Any,
+    fmt::{Display, Formatter},
+    fs, io,
+    path::{Path, PathBuf},
+    result,
+};
 
 use logger::warn;
 
 // Based on https://elixir.free-electrons.com/linux/v4.9.62/source/arch/arm64/kernel/cacheinfo.c#L29.
 const MAX_CACHE_LEVEL: u8 = 7;
 
 #[derive(Debug)]
-pub enum Error {
+pub(crate) enum Error {
     FailedToReadCacheInfo(io::Error),
     InvalidCacheAttr(String, String),
     MissingCacheConfig,
-    MissingOptionalAttr(String),
+    MissingOptionalAttr(String, CacheEntry),
 }
 
 type Result<T> = result::Result<T, Error>;
 
-#[derive(Clone)]
-pub(crate) struct CacheInfo {
+struct CacheEngine {
+    store: Box<dyn CacheStore>,
+}
```

Comment:
Do we need the inner store ? If we could implement a trait like `CacheStore` (probably renamed) directly on the cache engine, it would make the code simpler. We wouldn't need to use any `#[cfg(not(test))]` anymore, and also we wouldn't need `as_any`. In the actual implementation we would use the `HostCacheEngine` directly and in the tests we would use the `MockCacheEngine` directly.